### PR TITLE
CoinbasePro: Fix funds parameter for market orders

### PR DESCRIPTION
**Pull request by dsinuela-taurus**
# PR Description

This PR fixes an issue when placing Market orders with the CoinbasePro wrapper interface.

To trigger this issue the Submit struct needs to look like this:

```
gctPair, err := currency.NewPairFromString(pair.Symbol)
if err != nil {
    // ...
}
submit := order.Submit{
    Exchange:  "coinbasepro",
    Amount:    0.001,
    Type:      order.Market,
    Side:      order.Buy,
    AssetType: asset.Spot,
    Pair:      gctPair,
}
resp, err := e.SubmitOrder(ctx, &submit)
```

The code was passing `Amount` both as `size` and `funds` parameters. These two parameters are optional (one of them must be provided) but have different units. The exchange complained about the precision of "funds" being too accurate, while only the size should have been set.

Error:

`CoinbasePro unsuccessful HTTP status code: 400 raw response: {\"message\":\"funds is too accurate. Smallest unit is 0.01\"}`

Looking at the Submit struct and the code that uses it, it looks like the intended source for `funds` should be `QuoteAmount`.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Tested the fix from an external project that imports gocryptotrader's coinbasepro interface and uses it standalone. Tested against coinbasepro production server.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/coinbasepro/coinbasepro_test.go [hunk 1]
```diff
@@ -490,7 +490,7 @@ func areTestAPIKeysSet() bool {
 	return c.ValidateAPICredentials(c.GetDefaultCredentials()) == nil
 }
 
-func TestSubmitOrder(t *testing.T) {
+func TestSubmitLimitOrder(t *testing.T) {
```
Comment (gloriousCode) [inline exchanges/coinbasepro/coinbasepro_test.go:493]: Could you please keep this as `TestSubmitOrder`? We like to have the test name match the function being tested (though you'll find exceptions I'm sure 😄)

Then can you just append the scenarios of `TestSubmitMarketOrderFromAmount` and  `TestSubmitMarketOrderFromQuoteAmount` to the test? like as follows:

```go

func TestSubmitOrder(t *testing.T) {
	if areTestAPIKeysSet() && !canManipulateRealOrders {
		t.Skip("API keys set, canManipulateRealOrders false, skipping test")
	}
	// limit order
	var orderSubmission = &order.Submit{
		Exchange: c.Name,
		Pair: currency.Pair{
			Delimiter: "-",
			Base:      currency.BTC,
			Quote:     currency.USD,
		},
		Side:      order.Buy,
		Type:      order.Limit,
		Price:     1,
		Amount:    0.001,
		ClientID:  "meowOrder",
		AssetType: asset.Spot,
	}
	response, err := c.SubmitOrder(context.Background(), orderSubmission)
	if areTestAPIKeysSet() && (err != nil || response.Status != order.New) {
		t.Errorf("Order failed to be placed: %v", err)
	} else if !areTestAPIKeysSet() && err == nil {
		t.Error("Expecting an error when no keys are set")
	}

	// market order from amount
	orderSubmission = &order.Submit{
		Exchange: c.Name,
		Pair: currency.Pair{
			Delimiter: "-",
			Base:      currency.BTC,
			Quote:     currency.USD,
		},
		Side:      order.Buy,
		Type:      order.Market,
		Amount:    0.001,
		ClientID:  "meowOrder",
		AssetType: asset.Spot,
	}
	response, err = c.SubmitOrder(context.Background(), orderSubmission)
	if areTestAPIKeysSet() && (err != nil || response.Status != order.New) {
		t.Errorf("Order failed to be placed: %v", err)
	} else if !areTestAPIKeysSet() && err == nil {
		t.Error("Expecting an error when no keys are set")
	}

	// market order from quote amount
	orderSubmission = &order.Submit{
		Exchange: c.Name,
		Pair: currency.Pair{
			Delimiter: "-",
			Base:      currency.BTC,
			Quote:     currency.USD,
		},
		Side:        order.Buy,
		Type:        order.Market,
		QuoteAmount: 1,
		ClientID:    "meowOrder",
		AssetType:   asset.Spot,
	}
	response, err = c.SubmitOrder(context.Background(), orderSubmission)
	if areTestAPIKeysSet() && (err != nil || response.Status != order.New) {
		t.Errorf("Order failed to be placed: %v", err)
	} else if !areTestAPIKeysSet() && err == nil {
		t.Error("Expecting an error when no keys are set")
	}
}

```

However, if you still would like individual tests of the same function, could you rename them to `TestSubmitOrder{{descriptionScenario}}` eg `TestSubmitOrderLimitOrder`?

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thank you for that. I can confirm that [it matches the documentation](https://docs.cloud.coinbase.com/exchange/reference/exchangerestapi_postorders-1) as well as uses the appropriate struct property of `QuoteAmount`. Thanks for the pickup! Just one small thing from me

File: exchanges/coinbasepro/coinbasepro_test.go (hunk 1 above)
Comment (dsinuela-taurus) [reply exchanges/coinbasepro/coinbasepro_test.go]: I applied the changes, thanks for showing the suggestion in code.

And thanks for the project :)

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes! Note: I dont have funds on the exchange, review was done via checking the API and testing API calls to test validation:
`Order failed to be placed: CoinbasePro unsuccessful HTTP status code: 400 raw response: {"message":"Insufficient funds"}`

**Review by shazbert — APPROVED** [review]
utACK - Thank you. 

```
CoinbasePro attempt 1 request path: https://api.pro.coinbase.com/orders
CoinbasePro request header [Cb-Access-Timestamp]: [1668124125]
CoinbasePro request header [Cb-Access-Key]: [meow]
CoinbasePro request header [Cb-Access-Passphrase]: [bruhkid]
CoinbasePro request header [Content-Type]: [application/json]
CoinbasePro request header [Cb-Access-Sign]: [QbXZxXap9xqas+WOoRY0EBZjPV5BJdr2tusgp7fQqUc=]
CoinbasePro request type: POST
CoinbasePro request body: {"funds":"1","product_id":"BTC-USD","side":"buy","type":"market"}
```

**Review by thrasher- — APPROVED** [review]
utACK, no funds on exchange but code looks good. Thanks!

**Pull request merged**